sdk/src/main/java/com/mapbox/maps/StyleObserver.kt

```diff
@@ -83,21 +90,29 @@ internal class StyleObserver(
   }
 
   override fun onStyleDataLoaded(eventData: StyleDataLoadedEventData) {
-    // style data arrives in following order: STYLE, SOURCES, SPRITE
-    // transition options must be applied after style but before sprite and sources to take effect
-    loadStyleTransitionOptions?.let {
-      if (eventData.type == StyleDataType.STYLE) {
-        nativeMap.styleTransition = it
-        // per gl-native docs style transition options should be reset for a new style so resetting them here
-        loadStyleTransitionOptions = null
+    println("Style data : $eventData")
+    when (eventData.type) {
+      StyleDataType.STYLE -> {
+        loadedStyle?.markInvalid()
+        loadedStyle = Style(nativeMap, pixelRatio).also {
+          styleDataStyleLoadedListener?.onStyleLoaded(it)
+        }
+        styleDataStyleLoadedListener = null
+      }
+      StyleDataType.SPRITE -> {
+        styleDataSpritesLoadedListener?.onStyleLoaded(loadedStyle!!)
```

Comment:
we try to avoid `!!`

Reply:
in general to make it work as expected we need to be sure that `StyleDataType.STYLE` is always triggered before `StyleDataType.SPRITE` and `StyleDataType.SOURCES` - is this always the case?

Reply:
Yes, `StyleDataType.STYLE` must be always emitted first, before two other events.